Sources/Apodini/WebService.swift

```diff
@@ -34,15 +34,19 @@ extension WebService {
     public mutating func run() throws {
         try Self.start(webService: self)
     }
+    /// The command configuration of the `ParsableCommand`
+    public static var configuration: CommandConfiguration {
+        CommandConfiguration(subcommands: Self().configuration._commands)
+    }
     
     /// This function is executed to start up an Apodini `WebService`
     /// - Parameters:
     ///    - waitForCompletion: Indicates whether the `Application` is launched or just booted. Defaults to true, meaning the `Application` is run
-    ///    - webService: The instanciated `WebService` by the Swift ArgumentParser containing CLI arguments.  If `WebService` isn't already instantiated by the Swift ArgumentParser, automatically create a default instance
+    ///    - app: The instanciated `Application` that will be used to boot and start up the web service. Passes a default plain application, if nothing is specified.
+    ///    - webService: The instanciated `WebService` by the Swift ArgumentParser containing CLI arguments.  If `WebService` isn't already instanciated by the Swift ArgumentParser, automatically create a default instance
     /// - Returns: The application on which the `WebService` is operating on
     @discardableResult
-    static func start(waitForCompletion: Bool = true, webService: Self = Self()) throws -> Application {
-        let app = Application()
+    public static func start(waitForCompletion: Bool = true, app: Application = Application(), webService: Self = Self()) throws -> Application {
```

Comment:
Maybe consider replacing the `waitForCompletion` with an enum that has a bit clearer semantic meeting. I am not sure if the current naming is that great wither. Isn't the `run` method the one that is blocking and the the `waitForCompletion` call only indicates if the `LifecycleHandler` are called with?
- `.run`: Starts up the complete web service and blocks the execution until the web service exits itself or gets killed
- `.start`: Starts up the complete web service but doesn't block so the `run()` method is never called and the web service just exists (probably what we want for quite some things such as the exporters so they don't have to call `exit()`

With the possibility to add additional elements in the future.

Reply:
See my [comment](https://github.com/Apodini/Apodini/pull/326#discussion_r686803378) here